core/servers/content/web_handlers/webfinger.js

```diff
@@ -0,0 +1,135 @@
+const Log = require('../../../logger').log;
+
+exports.moduleInfo = {
+    name: 'WebFinger',
+    desc: 'A simple WebFinger Handler',
+    author: 'NuSkooler',
+    packageName: 'codes.l33t.enigma.web.handler.finger',
+};
+
+exports.getModule = class WebFingerServerModule extends ServerModule {
+    constructor() {
+        super();
+    }
+
+    init(cb) {
+        if (!_.get(Config(), 'contentServers.web.handlers.webFinger.enabled')) {
+            return cb(null);
+        }
+
+        const { getServer } = require('../../../listening_server');
+
+        // we rely on the web server
+        this.webServer = getServer(WebServerPackageName);
+        if (!this.webServer || !this.webServer.instance.isEnabled()) {
+            return cb(Errors.UnexpectedState('Cannot access web server!'));
+        }
+
+        this.webServer.instance.addRoute({
+            method: 'GET',
+            path: /^\/\.well-known\/webfinger\/?\?/,
+            handler: this._webFingerRequestHandler.bind(this),
+        });
+
+        return cb(null);
+    }
+
+    _webFingerRequestHandler(req, resp) {
+        const url = new URL(req.url, `https://${req.headers.host}`);
+
+        const resource = url.searchParams.get('resource');
+        if (!resource) {
+            return this.webServer.instance.respondWithError(
+                resp,
+                400,
+                '"resource" is required',
+                'Missing "resource"'
+            );
+        }
+
+        this._getUser(resource, resp, (err, user, accountName) => {
+            if (err) {
+                // |resp| already written to
+                return Log.warn({ error: err.message }, `WebFinger failed: ${req.url}`);
+            }
+
+            const body = JSON.stringify({
+                subject: `acct:${accountName}`,
+                links: [this._profilePageLink(user)],
+            });
+
+            const headers = {
+                'Content-Type': 'application/jrd+json',
+                'Content-Length': body.length,
+            };
+
+            resp.writeHead(200, headers);
+            return resp.end(body);
+        });
+    }
+
+    _profilePageLink(user) {
+        const href = this.webServer.instance.buildUrl(`/wf/@${user.username}`);
+        return {
+            rel: 'http://webfinger.net/rel/profile-page',
+            type: 'text/plain',
+            href,
+        };
+    }
+
+    _getUser(resource, resp, cb) {
+        // we only handle "acct:NAME" URIs
+
+        const notFound = () => {
+            this.webServer.instance.respondWithError(
+                resp,
+                404,
+                'Resource not found',
+                'Resource Not Found'
+            );
+        };
+
+        const acctIndex = resource.indexOf('acct:', 0);
+        if (0 != acctIndex) {
+            notFound();
+            return cb(Errors.DoesNotExist('"acct:" missing'));
+        }
+
+        const accountName = resource.substring(acctIndex + 5);
```

Comment:
acctIndex is always 0 - can just use resource.substring(5) here I believe

Reply:
We want to start past `acct:`